crates/wasi-common/src/snapshots/wasi_snapshot_preview1.rs

```diff
@@ -0,0 +1,1448 @@
+use crate::entry::{Descriptor, Entry};
+use crate::sandboxed_tty_writer::SandboxedTTYWriter;
+use crate::wasi::wasi_snapshot_preview1::WasiSnapshotPreview1;
+use crate::wasi::{types, AsBytes, Errno, Result};
+use crate::WasiCtx;
+use crate::{clock, fd, path, poll};
+use log::{debug, error, trace};
+use std::cell::Ref;
+use std::convert::TryInto;
+use std::io::{self, Read, Seek, SeekFrom, Write};
+use std::ops::DerefMut;
+use wiggle_runtime::{GuestBorrows, GuestPtr};
+
+impl<'a> WasiSnapshotPreview1 for WasiCtx {
+    fn args_get<'b>(
+        &self,
+        mut argv: GuestPtr<'b, GuestPtr<'b, u8>>,
+        mut argv_buf: GuestPtr<'b, u8>,
+    ) -> Result<()> {
+        trace!("args_get(argv_ptr={:?}, argv_buf={:?})", argv, argv_buf);
+
+        let mut argv_buf_offset: types::Size = 0;
+        let mut bc = GuestBorrows::new();
+
+        for arg in &self.args {
+            let arg_bytes = arg.as_bytes_with_nul();
+            let elems = arg_bytes.len().try_into()?;
+            let as_arr = argv_buf.as_array(elems);
+            let raw_ptr = as_arr.as_raw(&mut bc)?;
```

Comment:
I think the usage of `GuestBorrows` here technically may not be correct. There's no real need to do any borrow checking here because there's no long-lived borrows, this is effectively just a "given this guest slice pointer write in these bytes' which is in general a safe operation (like `GuestPtr::write` is)

Reply:
I guess you may be right, however, without the `GuestBorrows` I cannot obtain a mutable ref to `[u8]`, and there's not other simple way of copying the contents of `arg_bytes` into the `argv_buf` pointer currently. The only other way I can see is manually incrementing `argv_buf` pointer and copying byte-by-byte from `arg_bytes`. Unless I'm perhaps missing something here?

Reply:
Oh sorry my point is that persisting the same `GuestBorrows` for the entirety of this method I believe is not necessary. Each write can use an independent instance of `GuestBorrows` since there's no need to prevent overlap in this method.

Reply:
That's perfect! It'll clean up the code nicely! 😎